RRemoteConfig/ConfigCache.swift

```diff
@@ -1,31 +1,119 @@
-                self.activeConfig = ConfigModel(config: dictionary)
+
+                guard let configDic = dictionary["config"] as? [String: String] else {
+                    return
+                }
+                var configModel = ConfigModel(config: configDic, keyId: dictionary["keyId"] as? String ?? "")
+                configModel.signature = dictionary["signature"] as? String
+
+                self.verifyContents(model: configModel, resultHandler: { (verified) in
+                    if verified {
+                        print("Set active config to cached contents")
+                        self.activeConfig = configModel
+                    } else {
+                        print("Dictionary contents verification failed")
+                    }
+                })
             }
         }
     }
 
+    func refreshFromRemote() {
+        self.poller.start {
+            self.fetcher.fetchConfig { (result) in
+                guard let configModel = result else {
+                    return print("Config could not be refreshed from remote")
+                }
+                self.verifyContents(model: configModel, resultHandler: { (verified) in
+                    if verified {
+                        let dictionary = [
+                            "config": configModel.config,
+                            "keyId": configModel.keyId as Any,
+                            "signature": configModel.signature as Any
+                        ]
+                        self.write(dictionary)
+                    } else {
+                        print("Dictionary contents verification failed")
+                    }
+                })
+            }
+        }
+    }
+
+    fileprivate func write(_ config: [String: Any]) {
+        DispatchQueue.global(qos: .utility).async {
+            NSDictionary(dictionary: config).write(to: self.cacheUrl, atomically: true)
+            #if DEBUG
+            let readFromPlist = NSDictionary(contentsOf: self.cacheUrl)
+            print("Config written to url \(self.cacheUrl):\n\n \(String(describing: readFromPlist))")
+            #endif
+        }
+    }
+}
+
+extension FileManager {
+    class func getCacheDirectory() -> URL {
+        let cachePaths = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)
+        return cachePaths[0]
+    }
+}
+
+// MARK: Payload signature verification
+extension ConfigCache {
+    func verifyContents(model: ConfigModel, resultHandler: @escaping (Bool) -> Void ) {
+
+        // Fetch key from backend if not found in key store
+        if let key = keyStore.key(for: model.keyId) {
+            let verified = self.verifier.verify(signatureBase64: model.signature ?? "",
+                                                dictionary: model.config,
+                                                keyBase64: key)
+            resultHandler(verified)
+        } else {
+            fetcher.fetchKey(with: model.keyId) { (keyModel) in
```

Comment:
It fetches the key only if verification with the cached key fails, but you're right that it would delay the setting of the active config. The fallback values would get used until the config is verified with the newly fetched key. It's not predictable for the app to mix fallback and post-fetch config values so I'll change the approach to match android.